distribution/pull_v2_windows.go

```diff
@@ -23,16 +23,17 @@ func (ld *v2LayerDescriptor) Descriptor() distribution.Descriptor {
 }
 
 func (ld *v2LayerDescriptor) open(ctx context.Context) (distribution.ReadSeekCloser, error) {
-	if len(ld.src.URLs) == 0 {
-		blobs := ld.repo.Blobs(ctx)
-		return blobs.Open(ctx, ld.digest)
-	}
-
 	var (
 		err error
 		rsc distribution.ReadSeekCloser
 	)
 
+	// Try to get the layer from the registry.
+	blobs := ld.repo.Blobs(ctx)
+	if rsc, err = blobs.Open(ctx, ld.digest); err == nil {
+		return rsc, err
```

Comment:
@aaronlehmann: You're right about `Open` not returning an error if the blob doesn't exist. And I'm with you on `return rsc, nil`. 

How about this instead?

```go
	blobs := ld.repo.Blobs(ctx)
	rsc, err := blobs.Open(ctx, ld.digest)
        
	if len(ld.src.URLs) == 0 {
		return rsc, err
	}
        
	// We're done if the registry has this layer.
	if err == nil {
		if _, err = rsc.Seek(0, os.SEEK_SET); err == nil {
			return rsc, nil
		}
		rsc.Close()
	}
```